src/sentry/static/sentry/app/components/avatarChooser.tsx

```diff
@@ -7,79 +6,78 @@ import {t} from 'app/locale';
-  static propTypes = {
-    api: PropTypes.object,
-    endpoint: PropTypes.string.isRequired,
-    allowGravatar: PropTypes.bool,
-    allowLetter: PropTypes.bool,
-    allowUpload: PropTypes.bool,
-    type: PropTypes.oneOf(['user', 'team', 'organization', 'project']),
-    model: PropTypes.shape({
-      avatar: PropTypes.shape({
-        avatarType: PropTypes.oneOf(['upload', 'letter_avatar', 'gravatar']),
-      }),
-    }),
-    // Is this a chooser for a User account?
-    isUser: PropTypes.bool,
-    savedDataUrl: PropTypes.string,
-    onSave: PropTypes.func,
-    disabled: PropTypes.bool,
-  };
-
+import {Panel, PanelBody, PanelHeader} from 'app/components/panels';
+import Avatar from 'app/components/avatar';
+import AvatarCropper from 'app/components/avatarCropper';
+import Button from 'app/components/button';
+import ExternalLink from 'app/components/links/externalLink';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import {Client} from 'app/api';
+import {AvatarUser, Organization, Project, Team} from 'app/types';
+
+type Model = Pick<AvatarUser, 'avatar'>;
+type AvatarType = Required<Model>['avatar']['avatarType'];
+
+type Props = {
+  api: Client;
+  endpoint: string;
+  allowGravatar?: boolean;
+  allowLetter?: boolean;
+  allowUpload?: boolean;
+  type: 'user' | 'team' | 'organization' | 'project';
+  model: Model;
+  isUser: boolean;
+  savedDataUrl: string;
+  onSave: (model: Model) => void;
+  disabled: boolean;
+};
+
+type State = {
+  model: Model;
+  hasError: boolean;
+  savedDataUrl?: string | null;
+  dataUrl?: string | null;
+};
+
+class AvatarChooser extends React.Component<Props, State> {
   static defaultProps = {
     allowGravatar: true,
     allowLetter: true,
     allowUpload: true,
     onSave: () => {},
   };
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      model: this.props.model,
-      savedDataUrl: null,
-      dataUrl: null,
-      hasError: false,
-    };
-  }
+  state: State = {
+    model: this.props.model,
+    savedDataUrl: null,
+    dataUrl: null,
+    hasError: false,
+  };
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps: Props) {
     // Update local state if defined in props
     if (typeof nextProps.model !== 'undefined') {
       this.setState({model: nextProps.model});
     }
   }
 
-  updateState(model) {
+  updateState(model: Model) {
     this.setState({model});
   }
 
-  updateDataUrlState = dataUrlState => {
-    this.setState(dataUrlState);
-  };
-
-  handleError(msg) {
+  handleError(msg: string) {
     addErrorMessage(msg);
   }
 
-  handleSuccess(model) {
+  handleSuccess(model: Model) {
     const {onSave} = this.props;
     this.setState({model});
     onSave(model);
     addSuccessMessage(t('Successfully saved avatar preferences'));
   }
 
-  handleSaveSettings = ev => {
+  handleSaveSettings = (ev: React.MouseEvent) => {
```

Comment:
```suggestion
  handleSaveSettings = (ev: React.MouseEvent<HTMLButtonElement>) => {
```

Reply:
I'm fine with the type just being the less narrow `Element` since we don't use any of the `HTMLButtonElement` interface, thus if we ever changed what was doing the clicking, we wouldn't need to update this as long as it's assigned to an Element.